packages/eservice-template-process/src/services/validators.ts

```diff
@@ -109,3 +117,21 @@ export function assertPublishedEServiceTemplate(
     throw eserviceTemplateWithoutPublishedVersion(eserviceTemplate.id);
   }
 }
+
+export function hasAccessToDraftTemplateVersions(
+  authData: UIAuthData | M2MAuthData | M2MAdminAuthData
+): boolean {
+  // NOTE: this is not sufficient to access draft versions.
```

Comment:
may be worth to rename the function to avoid misunderstanding?
Something like `hasRoleToDraftTemplateVersions`